tools/pythonpkg/src/numpy/array_wrapper.cpp

```diff
@@ -572,26 +572,44 @@ static bool ConvertColumnRegular(NumpyAppendData &append_data) {
 }
 
 template <class DUCKDB_T>
-static bool ConvertDecimalInternal(NumpyAppendData &append_data, double division) {
+static PyObject *ConstructDecimal(DUCKDB_T value, uint8_t width, uint8_t scale) {
+	auto decimal_length = DecimalToString::DecimalLength<DUCKDB_T>(value, width, scale);
+
+	auto &python_import_cache = *DuckDBPyConnection::ImportCache();
+	auto decimal = python_import_cache.decimal.Decimal();
+
+	unique_ptr<char[]> raw_string(new char[decimal_length]);
```

Comment:
This looks medium hacky,`decimal_length` is variable based on the `value` so we can't reuse the buffers
We might be able to find the max length given the physical type and reuse the buffer that way.

Reply:
After seeing the enormous regression I figured I would reuse the string buffer and see how much that helps
From 808 -> 802 seconds, so.. not much

I'll have to profile it but I think the problem is just the constructor of the decimal.Decimal objects, to which there isn't much that can be done.

We could instead keep the old conversion and put this one behind a flag??
`python_preserve_decimal` ?